protocols/300073/300073.ot2.apiv2.py

```diff
@@ -0,0 +1,385 @@
+        """
+
+
+    '''
+    def transfer_tube_samples(volume, source_tubes, dest_wells):
+        nonlocal pipette, aspiration_height_mm, dispension_height_mm
+        nonlocal post_aspiration_wait
+
+        for s_well, d_well in zip(source_tubes, dest_wells):
+            pick_up(pipette)
+            pipette.aspirate(sample_aspiration_vol_ul,
+                             s_well.bottom(aspiration_height_mm))
+            ctx.delay(post_aspiration_wait)
+            pipette.dispense(sample_aspiration_vol_ul,
+                             d_well.bottom(dispension_height_mm))
+            pipette.drop_tip()
+
+    # reagents
+
+    '''
+    Define where all reagents are on the deck using the labware defined above.
+
+    e.g.
+
+    water = reservoir12.wells()[-1]
+    waste = reservoir.wells()[0]
+    samples = plate.rows()[0][0]
+    dnase = tuberack.wells_by_name()['A4']
+
+    '''
+
+    # plate, tube rack maps
+
+    '''
+    Define any plate or tube maps here.
+
+    e.g.
+
+    plate_wells_by_row = [well for row in plate.rows() for well in row]
+
+    '''
+    def add_quadrant_wells(list, columns, row_start_index, row_end_index):
+        """
+        Adds the wells in the given quadrant(rectangle) defined
+        by the columns and the start and end indices of the rows
+        col 1/row 1 ... col n/row 1
+        .  .            .
+        .       .       .
+        .            .  .
+        col 1/row n ... col n/row n
+        :param list: The list to append the wells in the quadrant to
+        :param colums: The columns that define the "x" interval of the quadrant
+        :param row_start_index: defines the start "y" interval
+        :param row_end_index: defines the end "y" interval
+        """
+        for col in columns:
+            for well in col[row_start_index:row_end_index]:
+                list.append(well)
+
+    # Map the target quadrants of the destination plate
+    targ_cols = plate.columns()
+    # Top left quadrant minus first 3 wells:
+    # The rectangle of A1-D1 to A6-D6 minus well A1 to A3
+    target_quadrant_1 = []
+    target_quadrant_1.append(targ_cols[0][3])  # Well D4
+    add_quadrant_wells(target_quadrant_1,
+                       targ_cols[1:6], 0, 4)
+
+    # Top right quadrant: A7-D7 to A12-D12 rectangle
+    target_quadrant_2 = []
+    add_quadrant_wells(target_quadrant_2,
+                       targ_cols[6:12], 0, 4)
+
+    # Bottom left quadrant: E1-H1 to E6-H6 rectangle
+    target_quadrant_3 = []
+    add_quadrant_wells(target_quadrant_3,
+                       targ_cols[0:6], 4, 8)
+
+    # Bottom right quadrant: E7-H7 to E12-H12 rectangle
+    target_quadrant_4 = []
+    add_quadrant_wells(target_quadrant_4,
+                       targ_cols[6:12], 4, 8)
+
+    target_well_map = []
+    for quadrant in [target_quadrant_1, target_quadrant_2, target_quadrant_3,
+                     target_quadrant_4]:
+        for well in quadrant:
+            target_well_map.append(well)
```

Comment:
interesting... so now flattening these quadrants... I think this could have been done with a nested for loop.

Reply:
I agree, but I thought maybe leaving it explicit would be helpful for readability.

Reply:
I agree, but maybe it's good for readability to leave things more explicit?

Reply:
👍